src/commands/CommandMusicPlay.java

```diff
@@ -35,7 +48,30 @@ public void action(){
 			}
 
 			if(getContent() != null){
-				MusicManager.get().loadTrack(this, getContent());
+				String id;
+				String source;
+				if (!isUrl(getContent())){
+					YouTube youtube = new YouTube.Builder(new NetHttpTransport(), new JacksonFactory(), new HttpRequestInitializer() {
+						public void initialize(HttpRequest request) throws IOException {
+						}
+					}).setApplicationName("Discord Bot").build();
+					try {
+						YouTube.Search.List search = youtube.search().list("snippet");
+						search.setMaxResults((long) 1);
+						search.setQ(getContent());
+						search.setKey(Environment.getVar("YOUTUBE_TOKEN"));
+						SearchListResponse response = search.execute();
+						id = response.getItems().get(0).getId().getVideoId();
+						System.out.print(id);
+						source = "https://www.youtube.com/watch?v=" + id;
+						MusicManager.get().loadTrack(this, source);
+					} catch (IOException e) {
+						e.printStackTrace();
```

Comment:
We'd need to handle exceptions where the bot catches this exception, for example, sending a message telling the user an error happened and to try again.

This would be a stretch but if it keeps failing (let's say after the second try), tell the user to try with the url of the video / music he wants to play. This would be a way better UX - IMO, of course.